### PR TITLE
Update Developer Guide for v1.3

### PR DESCRIPTION
**Pull request by macareonie**
The implementation section under the Developer Guide now includes description on the Find feature. #102

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -158,102 +158,59 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
 This section describes some noteworthy details on how certain features are implemented.
 
-### \[Proposed\] Undo/redo feature
+### Find feature
 
-#### Proposed Implementation
+#### Implementation
 
-The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+The find mechanism is facilitated by 3 classes `FindEmailCommand`, `FindNameCommand` and `FindPhoneCommand`. They all extend `FindCommand` with their corresponding `COMMAND_WORD` : `find_email`, `find_name` and `find_phone` respectively, as well as a corresponding `predicate` variable of type `EmailContainsKeywordsPredicate`, `NameContainsKeywordsPredicate` and `PhoneContainsKeywordsPredicate` respectively.
```
Comment (yashpola) [inline docs/DeveloperGuide.md:165]: So proud of you for this feature!

File: docs/diagrams/find/FindCommandClass.puml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+skinparam classAttributeIconSize 0
+
+Class "{abstract}\nCommand" as Command {
+  +execute(Model): CommandResult
+}
+Class FindCommand {
+  +MESSAGE_USAGE: String
+}
+Class FindEmailCommand {
+  +{static}COMMAND_WORD: String
```
Comment (yashpola) [inline docs/diagrams/find/FindCommandClass.puml:12]: Good job remembering to show class-level members

**Review by yashpola — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -158,102 +158,59 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
-### \[Proposed\] Undo/redo feature
+### Find feature
```
Comment (headcube1) [inline docs/DeveloperGuide.md:161]: Nice feature implemented

**Review by headcube1 — APPROVED** [review]
LGTM

**Pull request merged**